### PR TITLE
Check pip-compile and commit autoformatter changes

### PR DESCRIPTION
**Pull request by A5rocks**
I'm not sure having a commit afterwards is the best behavior for if someone is rapidly iterating, but this specifically would make dependabot PRs nicer to deal with. (dependabot has a nasty habit of removing things from the lockfile: https://github.com/python-trio/trio/pull/2511#discussion_r1058082186)

https://github.com/A5rocks/trio/actions/runs/4120613215/jobs/7115511631 proof this works

(I think, after this, the next step is adding an action that enables the auto-merge feature on any PR dependabot makes. After that: make dependabot update our github actions (ie `@actions/checkout` which ATM is out of date))

### CODE REVIEW

**Review by pquentin — APPROVED** [review]
Thanks! LGTM.

**Comment by A5rocks** [comment]
I was searching through history on the gitter channel and it turns out this has been mentioned before and turned down as too much a burden on a developer: https://gitter.im/python-trio/general?at=5d426fcee2802b6790a860be

I could make this use a specific PR comment to trigger instead. Any opinions? (ie `autoformat` on its own line?)

**Comment by pquentin** [comment]
Or maybe you could only apply this on dependabot PRs?

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -120,6 +120,14 @@ jobs:
           CHECK_FORMATTING: '${{ matrix.check_formatting }}'
           # Should match 'name:' up above
           JOB_NAME: 'Ubuntu (${{ matrix.python }}${{ matrix.extra_name }})'
+      - name: Commit autoformatter changes
+        continue-on-error: true
+        if: failure() && matrix.check_formatting == '1' && github.actor == 'dependabot[bot]'
+        run: |
+          git config user.name github-actions
+          git config user.email github-actions@github.com
```
[the remark this answers is not in the document]
Comment (A5rocks) [reply .github/workflows/ci.yml:128]: Huh, TIL! Thanks, I'll update this.

**Review by pquentin — APPROVED** [review]
Thanks! Looks good to me.

**Comment by A5rocks** [comment]
I had to commit https://github.com/python-trio/trio/commit/b1fbbd3b870a4acbc8ea14bb76ac6a4a2920ed22 (and a few other things) to make this work on dependabot PRs.

I'm not so sure that's secure; it certainly seems like a bad idea. Are these secure?

Looking at the docs (<https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#permissions>) says that this will only grant it if the actor has write permissions (or if we enabled a setting we probably didn't), so I suppose we're fine?

**Pull request merged**